nilearn/plotting/tests/test_surf_plotting.py

```diff
@@ -197,9 +197,141 @@ def test_plot_surf_roi_error():
     mesh = generate_surf()
     rng = np.random.RandomState(0)
     roi_idx = rng.randint(0, mesh[0].shape[0], size=5)
-
-    # Wrong input
     with pytest.raises(
             ValueError,
             match='roi_map does not have the same number of vertices'):
         plot_surf_roi(mesh, roi_map=roi_idx)
+
+
+def _generate_img():
+    mni_affine = MNI152TEMPLATE.get_affine()
+    data_positive = np.zeros((7, 7, 3))
+    rng = np.random.RandomState(42)
+    data_rng = rng.rand(7, 7, 3)
+    data_positive[1:-1, 2:-1, 1:] = data_rng[1:-1, 2:-1, 1:]
+    nii = nibabel.Nifti1Image(data_positive, mni_affine)
+    return nii
+
+
+def test_plot_img_on_surf_hemispheres_and_orientations():
+    nii = _generate_img()
+    # Check that all combinations of 1D or 2D hemis and orientations work.
+    plot_img_on_surf(nii, hemispheres='right', views='lateral')
+    plot_img_on_surf(nii, hemispheres=['left', 'right'], views='lateral')
+    plot_img_on_surf(nii,
+                     hemispheres='right',
+                     views=['medial', 'lateral'])
+    plot_img_on_surf(nii,
+                     hemispheres=['left', 'right'],
+                     views=['dorsal', 'medial'])
+
+
+def test_plot_img_on_surf_colorbar():
+    nii = _generate_img()
+    plot_img_on_surf(nii, hemispheres='right', views='lateral',
+                     colorbar=True, vmax=5, threshold=3)
+    plot_img_on_surf(nii, hemispheres='right', views='lateral',
+                     colorbar=False)
+    plot_img_on_surf(nii, hemispheres='right', views='lateral',
+                     colorbar=False, cmap='roy_big_bl')
+    plot_img_on_surf(nii, hemispheres='right', views='lateral',
+                     colorbar=True, cmap='roy_big_bl', vmax=2)
+
+
+def test_plot_img_on_surf_inflate():
+    nii = _generate_img()
+    plot_img_on_surf(nii, hemispheres='right', views='lateral',
+                     inflate=True)
+
+
+def test_plot_img_on_surf_surf_mesh():
+    nii = _generate_img()
+    plot_img_on_surf(nii, hemispheres=['right', 'left'], views='lateral',
+                     surf_mesh=None)
+    plot_img_on_surf(nii, hemispheres=['right', 'left'], views='lateral',
+                     surf_mesh='fsaverage')
```

Comment:
please remove the tests using 'fsaverage' because it needs to be downloaded from the internet. ideally we would mock fetching fsaverage but I don't think it's worth the trouble here, you can just keep fsaverage5

Reply:
Done.